### ISSUE TITLE
Breakpad Linux support

### ISSUE THREAD

**Issue by fluffyfreak** [pioneerspacesim/pioneer#3690]
It looks like adding support for [google breakpad to a Linux build](https://chromium.googlesource.com/breakpad/breakpad/+/master/docs/linux_starter_guide.md) should be fairly easy.

This would give it feature parity with the Windows build and might arguably be more useful.

**Comment by mike-f1** [comment]
@fluffyfreak , @impaktor give me an hint for this, so I think I could take care of this but I need some help...

First: in #3211 you mention the library is already installed... but where is it? I was unable to find it.

Could you briefly (and "as you talk with a child") explain how it's implemented right now for Windows?

**Comment by fluffyfreak** [comment]
Well the original PR is #3211 which includes all of the windows side files so you can see how I did it there :)

**Comment by fluffyfreak** [comment]
The actual breakpad files are in the `pioneer-thirdparty` folders.

**Comment by fluffyfreak** [comment]
@mike-f1 what OS do you develop on? I assume that it's a Linux distro yeah?

There's mulitple ways to do this I guess.
- move the breakpad stuff into `pioneer` under the `contrib` folder
- move the breakpad folder from `win32/include/` into `source`
- create a separate breakpad folder in `source` (**do not do this please**)

Probably best to discuss this in IRC for a bit.

**Comment by fluffyfreak** [comment]
For now though, do whatever you feel like doing, it's quite easy to move the folders or to change the VS2015 project to work with the new layout later on.

**Comment by mike-f1** [comment]
yes, I have tried the second, but what I didn't realize is that breakpad files aren't from "master" of its project.

Next step is to copy-paste folders (as you suggest) and verify if they compile under Linux.
Then I probably will smash on other problems

For now, I would only report that I did a make on pioneer-thirdparty and this blown my pc:
now I'm unable to compile pioneer because it says:
```
In file included from ../../src/vector3.h:9:0,
                 from JsonUtils.h:8,
                 from JsonUtils.cpp:8:
../../src/FloatComparison.h:7:24: fatal error: SDL_stdinc.h: No such file or directory
 #include <SDL_stdinc.h>

```
even if this file is in /usr/include/SDL  :-P

**Comment by mike-f1** [comment]
Ok, next problem: copy-paste don't works, because of lack of all autotools chain...

Due to my small knowledge I see only one way: I would put breakpad project files on
 `source/breakpad` as an "external" library (as for Assimp), but it belongs from Google
Git, so I don't know how to do such a thing... @fluffyfreak , my doubts are on how
to do it in a reliable way, the code itself I don't think will be a great problem
(...or at least I hope it)...

What do you think I have to do?

**Comment by mike-f1** [comment]
ok, I did some steps, but I'm even stuck: I need a way to specify a base include dir ( -Ipath/to/include ) for
breakpad through all the autotools chain :-P

More: there's an "optional" thing I left to do on autotools: include breakpad in the list of Makefile as
for Miniz, Json and so on.

...Basically I'm at the beginning, but no problem: I pin these things here only for myself

**Comment by mike-f1** [comment]
@fluffyfreak  and @impaktor : I made progress on this, but I'm not the best one to do "autotools stuffs" :/

On the link above I wrote some comments on what remain to do...
